test/file.jl

```diff
@@ -1156,3 +1156,35 @@ if !Sys.iswindows()
         test_22566()
     end
 end  # !Sys.iswindows
+
+function test_22922()
+    mktempdir() do tmpdir
+        (path, dirname) = splitdir(tmpdir)
+        @test dirname[1:3] == "jl_"
+    end
+    mktempdir(;prefix="ABC") do tmpdir
+        (path, dirname) = splitdir(tmpdir)
+        @test dirname[1:3] == "ABC"
+    end
+    mktemp() do tmp,io
+        (path, filename) = splitdir(tmp)
+        @test filename[1:3] == "jl_"
+    end
+    mktemp(;prefix="ABC") do tmp,io
+        (path, filename) = splitdir(tmp)
+        @test filename[1:3] == "ABC"
+    end
+    tmp = tempname()
+    (path,filename) = splitdir(tmp)
+    @test filename[1:3]=="jl_"
+
+    tmp = tempname(;prefix = "0123456789")
+    (path,filename) = splitdir(tmp)
+    @test filename[1:3] == "012"
+    if !Sys.iswindows()
+        #Unix like OS have prefix size limit of 5 chars
```

Comment:
http://pubs.opengroup.org/onlinepubs/9699919799/functions/tempnam.html

Many applications prefer their temporary files to have certain initial letter sequences in their names. The pfx argument should be used for this. This argument may be a null pointer or point to a string of up to five bytes to be used as the beginning of the filename.